src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -164,3 +164,11 @@ createFxAdjustedPrincipalClaim dateToTime ownerReceives fxRateMultiplier notiona
   in
     prepareAndTagClaims dateToTime [fxLegClaim] "Principal payment"
 -- FIXED_RATE_BOND_REDEMPTION_CLAIM_END
+
+-- | Scale a claim by an asset performance
+scaleByAssetPerformance : Date -> Date -> O -> C -> C
+scaleByAssetPerformance startDate endDate asset claim =
+  when (TimeGte startDate) $ scale (Const 1.0 / asset) $
+  when (TimeGte endDate) $ scale (asset) $
+  claim
```

Comment:
I would change this to include the correction claim. I think it makes sense then for the `Asset Swap` to use this utility function as well

Reply:
just realise that this is already used by the asset swap performance :)